pinot-spi/src/main/java/org/apache/pinot/spi/config/provider/PinotConfigProvider.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.config.provider;
+
+import org.apache.pinot.spi.config.table.TableConfig;
+import org.apache.pinot.spi.data.Schema;
+
+
+/**
+ * An interface for the provider of pinot table configs and schema
+ */
+public interface PinotConfigProvider {
+
+  /**
+   * Gets the tableConfig
+   */
+  TableConfig getTableConfig(String tableNameWithType);
+
+  /**
+   * Registers the {@link TableConfigChangeListener} and notifies it whenever any changes (addition, update, removal)
+   * to any of the table configs are detected
+   */
+  void registerTableConfigChangeListener(TableConfigChangeListener tableConfigChangeListener);
```

Comment:
Let us say a user of this interface wants to track the changes in table config. Looks like they need to get table config first and then register to listen to changes.

Is there a race condition here? (changes may happen before they register, for instance).

Does'nt it make sense to return the table config at the time of registration?

Same with schema I suppose

Reply:
They can directly register the listener. No need to get table config first.

Reply:
On second thought, that does make it better for the specific case you're talking about. changed it.